### PR TITLE
statistics: merge global stats even if some partition stats are missing

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: ref #38999

Problem Summary:

### What is changed and how it works?

Introduce a system variable `tidb_skip_missing_partition_stats`, whose default value is `on`. If `tidb_skip_missing_partition_stats` turns on, merging global stats would skip partitions whose stats are missing rather than fail.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
@xuyifangreeneyes Please fix the conflict.

File: statistics/handle/handle.go [hunk 1]
```diff
@@ -741,36 +760,52 @@ func (h *Handle) mergePartitionStats2GlobalStats(sc sessionctx.Context,
...
+				err = err1
 				return
 			}
 			if allPartitionStats == nil {
 				allPartitionStats = make(map[int64]*statistics.Table)
 			}
 			allPartitionStats[partitionID] = partitionStats
 		}
+		globalStats.Count += partitionStats.Count
+		globalStats.ModifyCount += partitionStats.ModifyCount
```
Comment (chrysan) [inline statistics/handle/handle.go:779]: Do we need to remove the count adding later in this func?

**Review by chrysan — COMMENTED** [review]
please add tests

File: statistics/handle/handle.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply statistics/handle/handle.go:779]: Yes. It is a mistake I made when resolving conflicts.

**Review by qw4990 — APPROVED** [review]

**Comment by qw4990** [comment]
Please fix the UT failure:
```
    result.go:49: 
        	Error Trace:	testkit/result.go:49
        	            				executor/test/analyzetest/analyze_test.go:2802
        	Error:      	Not equal: 
        	            	expected: "[Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1]\n[Warning 8244 Build global-level stats failed due to missing partition-level column stats: table `t` partition `p0` column `d`, please run analyze table to refresh columns of all partitions]\n"
        	            	actual  : "[Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1]\n"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,2 @@
        	            	 [Note 1105 Analyze use auto adjusted sample rate 1.000000 for table test.t's partition p1]
        	            	-[Warning 8244 Build global-level stats failed due to missing partition-level column stats: table `t` partition `p0` column `d`, please run analyze table to refresh columns of all partitions]
        	            	 
        	Test:       	TestAnalyzePartitionStaticToDynamic
```

**Review by hawkingrei — APPROVED** [review]

**Review by chrysan — APPROVED** [review]

File: statistics/handle/ddl.go [hunk 2]
```diff
@@ -161,8 +163,16 @@ func (h *Handle) updateGlobalStats(tblInfo *model.TableInfo) error {
 	if err != nil {
 		return err
 	}
+	if len(newColGlobalStats.MissingPartitionStats) > 0 {
+		logutil.BgLogger().Warn("missing partition stats when merging global stats", zap.String("table", tblInfo.Name.L),
```
Comment (chrysan) [inline statistics/handle/ddl.go:167]: What about adding it to statement warning message too?

File: statistics/handle/ddl.go [hunk 3]
```diff
@@ -192,8 +202,16 @@ func (h *Handle) updateGlobalStats(tblInfo *model.TableInfo) error {
 		if err != nil {
 			return err
 		}
+		if len(newIndexGlobalStats.MissingPartitionStats) > 0 {
+			logutil.BgLogger().Warn("missing partition stats when merging global stats", zap.String("table", tblInfo.Name.L),
```
Comment (chrysan) [inline statistics/handle/ddl.go:206]: ditto

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**